utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -117,6 +117,44 @@
 //!
 //! Note that *none* of the try-runtime operations need unsafe RPCs.
 //!
+//! ## Migration Best Practices
+//!
+//! One of the main use-cases of try-runtime is using it for testing storage migrations. The
+//! following points makes sure you can *effectively* test your migrations with try-runtime.
+//!
+//! #### Adding pre/post hooks
+//!
+//! One of the gems that come only in the `try-runtime` feature flag is the `pre_upgrade` and
+//! `post_upgrade` hooks for [`OnRuntimeUpgrade`]. This trait is implemented either inside the
+//! pallet, or manually in a runtime, to define a migration. In both cases, these functions can be
+//! added, given the right flag:
+//!
+//! ```ignore
+//! #[cfg(feature = try-runtime)]
+//! fn pre_upgrade() -> Result<(), &'static str> {}
+//!
+//! #[cfg(feature = try-runtime)]
+//! fn post_upgrade() -> Result<(), &'static str> {}
+//! ```
+//!
+//! (The pallet macro syntax will support this simply as a part of `#[pallet::hooks]`).
+//!
+//! These hooks allow you to execute some code, only within the `on-runtime-upgrade` command, before
+//! and after the migration. If any data needs to be temporarily stored between the pre/post
+//! migration hooks, [`OnRuntimeUpgradeHelpersExt`] can help with that.
+//!
+//! #### Logging
+//!
+//! It is super helpful to make sure your migration code uses logging (always with a `runtime` log
+//! target prefix, e.g. `runtime::balance`) and state exactly at which stage it is, and what it is
+//! doing.
+//!
+//! #### Guarding migrations
+//!
+//! Always make sure that any migration code is guarded either by [`StorageVersion`], or by some
+//! custom storage item, so that it is NEVER executed twice, even if the code lives in two
+//! consecutive runtimes.
```

Comment:
maybe we could have a migration best practice which state general best practice and link to all tools to use such as try-runtime. I feel like this best practices about guarding is more general than try-runtime.
That said it is good as it is, and better than what we have currently so we should ship this PR

Reply:
yeah I totally agree, I initially wanted to draft a new document in `maintain/MIGRATION.md`, but that would take more time. For now using `try-runtime` as the point of reference about migration is okay. Gradually, it would move elsewhere (potentially to devhub).